notebooks/compute-network-v2.ipynb

```diff
@@ -0,0 +1,464 @@
+    "\n",
+    "# clip the GeoDataFrame to include ONLY segments in the specified river network\n",
+    "network_segs = subreachable[tw]\n",
+    "network_gdf = gdf.loc[gdf.index.isin(network_segs)]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create an animation showing flow changes through space and time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gif\n",
+    "from mpl_toolkits.axes_grid1 import make_axes_locatable\n",
+    "\n",
+    "# network geodata framee\n",
+    "network_gdf = gdf.loc[gdf.index.isin(network_segs)]\n",
+    "\n",
+    "@gif.frame\n",
+    "def plot(tstep):\n",
+    "    \n",
+    "    first_flow = network_flows.loc[:, ((tstep), 'q')]\n",
+    " \n",
+    "    joined_gdf = network_gdf.join(first_flow)\n",
+    "    joined_gdf = joined_gdf.rename(columns = {joined_gdf.columns[-1] : 'flow'})\n",
+    "    \n",
+    "    # create a figure showing spatial variations in simulated flow and the lateral flow loading\n",
+    "    fig = plt.figure(constrained_layout=True)\n",
+    "    gs = fig.add_gridspec(4, 3)\n",
+    "    ax1 = fig.add_subplot(gs[0:2, :]) # we will make a map on ax1\n",
+    "    ax2 = fig.add_subplot(gs[2:3,:])  # we will make a line plot on ax2\n",
+    "\n",
+    "    # need to adjust the size and position of the map colorbar\n",
+    "    divider = make_axes_locatable(ax1)\n",
+    "    cax = divider.append_axes(\"right\", size=\"5%\", pad=0.1)\n",
+    "\n",
+    "    # plot the spatial variation in simulated flow rate, across all stream segments in the network\n",
+    "    joined_gdf.plot(markersize = 5, \n",
+    "                    column = \"flow\", \n",
+    "                    cmap = 'PuBu', \n",
+    "                    legend = True, \n",
+    "                    vmin = 0, \n",
+    "                    vmax = 20000,\n",
+    "                    ax = ax1, \n",
+    "                    cax = cax, \n",
+    "                    legend_kwds={'label': \"Flow (cms)\",\n",
+    "                             'orientation': \"vertical\"})\n",
+    "\n",
+    "    # map title and axis controls\n",
+    "    ax1.set_title(\"Mississippi River Basin\", size = 20)\n",
+    "    ax1.axis('off')\n",
+    "\n",
+    "    # plot the timeseris of lateral inflow loading\n",
+    "    qlats.iloc[1][0:tstep].plot(ax = ax2, color = 'r', linewidth = 2) # ! make the 5 = tstep\n",
+    "    ax2.set_xlim([0, 8000])\n",
+    "    ax2.set_ylim([0, 11])\n",
+    "    ax2.set_title(\"Lateral Inflow Loading\", size = 20)\n",
+    "    ax2.set_ylabel(\"Lateral Inflow (cms per node)\", size = 14)\n",
+    "    ax2.set_xlabel(\"Simulated Timestep\", size = 14)\n",
+    "\n",
+    "    # adjsut the figure size\n",
+    "    fig.set_size_inches(10, 10)\n",
+    "\n",
+    "frames = []\n",
+    "for i in range(0,8000,100):\n",
+    "    \n",
+    "    frame = plot(i)\n",
```

Comment:
Use a reference to the above entered time step numbers and lengths, etc. here.

Reply:
Definitely. This is now changed